src/components/map-overlays/map-overlays.component.tsx

```diff
@@ -21,9 +22,10 @@ interface IProps {
   tappedBuilding: BuildingId;
   zoomLevel: ZoomLevel;
   indoorInformation: IndoorInformation;
-  setMarkerLocation: (poi: POI) => void;
-  endLocation: POI;
-  startLocation: MarkerLocation;
+  setPOIMarkerLocation: (poi: POI) => void;
+  setBuildingMarkerLocation: (building: Building) => void;
+  endLocation: POI | Building;
```

Comment:
If possible, endLocation should be of type `MarkerLocation | Building`, to be consistent with the start location. If doing this brings too many complications, then just set startLocation as `POI | Building`

Also, another minor thing, put the declaration of startLocation above endLocation, for consistency :)

Reply:
Modified `endLocation: MarkerLocation | Building` and put `startLocation` before `endLocation`

Reply:
Thanks !